streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -439,4 +453,9 @@ public void flushState() {
         recordCollector.flush();
     }
 
+    // for testing only
+    Producer<byte[], byte[]> producer() {
+        return ((RecordCollectorImpl) recordCollector).producer();
```

Comment:
This cast seems suboptimal. Seems like it would be better to return the `recordCollector` and let the test or a test utility cast, if necessary. Non-test code shouldn't be assuming a particular implementation IMO (yes, I now that the method is for testing only, but it's better to limit such methods to providing access only).